Update Experiment docs for default deployment (#203)
amplitude/amplitude-dev-center: docs/experiment/general/data-model.md

## Deployments

In Amplitude Experiment, a deployment serves a group of flags or experiments for use in an application. Each [project](#projects) has a deployment using the project API key as the deployment key, available by default. On creation, experiment deployments have an associated randomly generated **deployment key** which Experiment uses to identify the deployment and authorize requests to the evaluation servers.

!!!info "Client vs Server Deployments"

>     Deployments are either client or server deployments. Client-side deployments should be used to initialize client-side SDKs, and server-side deployments should be used to initialize server-side SDKs or authorize requests to the Evaluation API.

Reviewer: ```suggestion
    Deployments are either client or server deployments. Use client-side deployments should to initialize client-side SDKs, and server-side deployments to initialize server-side SDKs or authorize requests to the Evaluation API.
```

>     Deployments are either client or server deployments. Client-side deployments should be used to initialize client-side SDKs, and server-side deployments should be used to initialize server-side SDKs or authorize requests to the Evaluation API.

Reviewer 2: ```suggestion
    Deployments are either client or server deployments. Use client-side deployments to initialize client-side SDKs, and server-side deployments to initialize server-side SDKs or authorize requests to the Evaluation API.
```

> Deployments live within Amplitude Analytics projects, and a project can have multiple deployments. Deployments are generally named after the platform (client-side) or service (server-side) which variants are served to (for example: `android`, `ios`, `web`). The default project API key deployment is useful for getting started, however using explicit deployments for each platform or service is the best practice for larger organizations or teams that may share the same Amplitude project across multiple platforms for the same application.

Reviewer: ```suggestion
Deployments belong to Amplitude Analytics projects, and a project can have multiple deployments. Amplitude recommends that you name deployments after the platform (client-side) or service (server-side) to which Experiment serves variants (for example: `android`, `ios`, `web`). The default project API key deployment is useful for getting started, but using explicit deployments for each platform or service is the best practice for larger organizations or teams that may share the same Amplitude project across multiple platforms for the same application.
```
